### PR TITLE
[Python] Concept extraction / references mega merge

### PR DESCRIPTION
**Pull request by yawpitch**
This branch pulls together all the various resource extraction exercises for a collective cleanup and harmonization before final merge.

No such work actually done as of right now, but this will allow anyone that wants to to track progress and comment.

### CODE REVIEW

**Comment by yawpitch** [comment]
Step one (tomorrow) will be getting all those blasted dead links fixed. ;-)

**Comment by yawpitch** [comment]
@BethanyG there's two files of yours ([markdown.py](https://github.com/exercism/v3/blob/python-concept-references/languages/python/reference/exercise-concepts/markdown.md) and [matrix.py](https://github.com/exercism/v3/blob/python-concept-references/languages/python/reference/exercise-concepts/matrix.md) that now both have **TODO** entries added ... you've linked to exercise examples that are behind a username, and so we can't actually link through to see them. Can you please copy and paste the body of each into the appropriate locations, inside triple-backspace code blocks?

If you can then create a PR from your fork back to this branch (**NOT** master)?

Thanks.

**Comment by yawpitch** [comment]
@ErikSchierboom can you take a look at approving this? I'd like to merge this into master and have everyone continue work from there in their own forks.

Closes #656
Closes #460 
Closes #459 
Closes #419 
Closes #417 
Closes #394 
Closes #392 
Closes #390 
Closes #389 
Closes #388 
Closes #387 
Closes #386

File: languages/python/reference/README.md [hunk 1]
```diff
@@ -1,5 +1,598 @@
...
+- [struct](../../../reference/types/struct.md)
+- [tuple](../../../reference/types/tuple.md)
+
+## Resources used
+
+- https://www.python.org
+
+## Extracted Concepts
+
+### [Dictionary][dictionary]
```
Comment (ErikSchierboom) [inline languages/python/reference/README.md]: The headers don't point to any resource. Was that intentional?

File: languages/python/reference/README.md [hunk 2]
```diff
@@ -1,5 +1,598 @@
-# Python reference
+# Python
```
[the remark this answers is not in the document]
Comment (yawpitch) [reply languages/python/reference/README.md]: Fair enough, I'll add that.

File: languages/python/reference/README.md (hunk 1 above)
Comment (yawpitch) [reply languages/python/reference/README.md]: Yeah, for now ... my plan is for all of these to be moved into discrete files, either they're general concepts that need to referenced in [REPO]/reference/concepts or they're truly Python-specific and need to be referenced in [LANGUAGE]/reference/concepts.  We'll do the work of divying them up from the merged master.

Comment (ErikSchierboom) [reply languages/python/reference/README.md]: > my plan is for all of these to be moved into discrete files

Great idea!

**Comment by ErikSchierboom** [comment]
@yawpitch p.s. you can check the "Convert existing files to new repository structure" item in the https://github.com/exercism/v3/blob/master/languages/python/README.md file.

**Comment by yawpitch** [comment]
> @yawpitch p.s. you can check the "Convert existing files to new repository structure" item in the https://github.com/exercism/v3/blob/master/languages/python/README.md file.

Will do when I'm a little more sure of the work.

**Comment by ErikSchierboom** [comment]
Okay!

**Pull request merged**